geometry/eolearn/geometry/superpixel.py

```diff
@@ -0,0 +1,88 @@
+"""
+Module for super-pixel segmentation
+"""
+
+import logging
+
+import skimage.segmentation
+import numpy as np
+
+from eolearn.core import EOTask, FeatureType, FeatureTypeSet
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SuperpixelSegmentation(EOTask):
+    """ Super-pixel segmentation task
+
+    Given a raster feature it will segment data into super-pixels. Representation of super-pixels will be returned as
+    a mask timeless feature where all pixels with the same value belong to one super-pixel
+    """
+    def __init__(self, feature, superpixel_feature, *, segmentation_object=skimage.segmentation.felzenszwalb,
```

Comment:
Why is there a * (additional positional arguments, if I understand correctly) in the `__init__`? They are never used, and never passed onwards. If this is to be future proof, then at least make them `*args` or so.

Reply:
`*` doesn't specify any new parameters, it is the way how to force keyword-only arguments. In this case it means that some can do
```Python
SuperpixelSegmentation(my_feature, my_superpixel_feature,
    segmentation_object=my_segmentation_object)
```
but can't do
```Python
SuperpixelSegmentation(my_feature, my_superpixel_feature,
    my_segmentation_object)
```
This way you leave an option that more keyword-only arguments might be added in the future and their order might change. Because you force users to write parameter names, their code will be more consistent and won't break because of such changes.
Some reference: https://stackoverflow.com/questions/2965271/forced-naming-of-parameters-in-python